Sources/XMTP/ApiClient.swift

```diff
@@ -105,9 +146,22 @@ class GRPCApiClient: ApiClient {
 	func subscribe(topics: [String]) -> AsyncThrowingStream<Envelope, Error> {
 		return AsyncThrowingStream { continuation in
 			Task {
-                for try await envelopeJson in self.rustClient.subscribe(topics: topics) {
-                    let envelope = try Envelope(jsonString: envelopeJson)
-					continuation.yield(envelope)
+				let topicsVec = RustVec<RustString>()
+				for topic in topics {
+					topicsVec.push(value: topic.intoRustString())
+				}
+				let subscription = try await self.rustClient.subscribe(topicsVec)
```

Comment:
One thing worth calling out is that I don't do any sort of retries in the Rust layer. Drive through a tunnel, deploy the nodes, whatever: the subscription will end. I've been assuming the caller would handle retries, but if that's not viable we can move the logic into Rust.